### ISSUE TITLE
Remove GrainService and provide a better alternative

### ISSUE THREAD

**Issue by ReubenBond** [dotnet/orleans#7103]
GrainServices are cumbersome and undocumented. They derive from SystemTarget, which is meant only for internal services and was not meant to be publicly exposed. Hence, I think we should remove them.

A preferable alternative would be to have regular grains which have fixed, per-silo placement.
Such grains can still include consistent hash rings and a similar kind of "grain service client" which maps the caller's grain id to that hash ring to find the correct grain service can also be implemented.

**Comment by ReubenBond** [comment]
Hi @buzzers, I agree. That can be accomplished using custom placement today, but I believe we should include a custom placement provider to do this within Orleans itself.

**Comment by gufigueiredo** [comment]
This issue is still valid? I'm wondering if there is a better way to fetch data (like a Repository) from grains. Is GrainServices still the correct approach?

**Comment by ReubenBond** [comment]
This issue is still valid - I still want an alternative to GrainServices.
For fetching data for grains, I would personally register & inject a regular singleton service like you typically would, rather than use a grain service.

**Comment by buzzers** [comment]
I was wondering if we could take a legacy actor model like akka and build virtual actors based on it.
In addition, I wonder if it is possible to specify that some nodes do not enable grain directory. This is used to reduce the number of actor reactivations when a node restarts.

**Comment by jonathansant** [comment]
Would the "fixed, per-silo placement" grain be long-lived like a grain service? Would you be able to have PeriodicTimers that can live as long as the silo lives?

**Comment by ReubenBond** [comment]
Yes, using Grains with fixed placement is better. Codifying this pattern would be useful, similar to the existing grain services pattern.

**Comment by bxjg1987** [comment]
Has there been any progress on this feature? It is difficult to understand the official documentation on grain service.
